File: php/src/RandomColors.php
<?php

namespace Vestaboard\Vbml;

/**
 * Port of randomColors.ts
 */
class RandomColors
{
    public static array $colorCodes = [63, 64, 65, 66, 67, 68, 69, 70];

    public static function generate(
        int $rows = 6,
        int $columns = 22,
        array $colors = []
    ): array {
        if (empty($colors)) {
            $colors = self::$colorCodes;
        }
        $board = [];
        for ($r = 0; $r < $rows; $r++) {
            $row = [];
            for ($c = 0; $c < $columns; $c++) {
                $row[] = $colors[array_rand($colors)];
            }
            $board[] = $row;
        }
        return $board;
    }
}

File: python/pyvbml/random_colors.py
"""Random colors.

Port of Vestaboard/vbml/src/randomColors.ts
"""

from __future__ import annotations

import random

from .character_codes import COLOR_CODES, CharacterCode
from .const import DIMENSIONS_FLAGSHIP


def random_colors(
    rows: int = DIMENSIONS_FLAGSHIP[0],
    columns: int = DIMENSIONS_FLAGSHIP[1],
    colors: list[CharacterCode] | list[int] | None = None,
) -> list[list[int]]:
    """Random colors."""
    if colors is None:
        colors = COLOR_CODES
    return [[int(random.choice(colors)) for _ in range(columns)] for _ in range(rows)]
